src/csharp/Microsoft.Spark/RDD/Collector.cs

```diff
@@ -83,5 +86,16 @@ public object Deserialize(Stream stream, int length)
                 return SerDe.ReadString(stream, length);
             }
         }
+
+        /// <summary>
+        /// Deserializer for Pickled Rows.
+        /// </summary>
+        private sealed class RowDeserializer : IDeserializer
+        {
+            public object Deserialize(Stream stream, int length)
+            {
+                return PythonSerDe.GetUnpickledObjects(stream, length);
```

Comment:
Can you add a comment where `RowCollector.GetRow()` is called?

Reply:
`RowConstructor.GetRow()` is now called from the `RowDeserializer` with a comment regarding `AutoBatchedPickler` which determines number of Rows being pickled.